src/components/Dialogs/DistanceMeasuringDialog/DistanceMeasuringDialog.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from "react";
+import {observable, action, makeObservable} from "mobx";
+import {observer} from "mobx-react";
+import {ColorResult} from "react-color";
+import {FormGroup, IDialogProps, NonIdealState, H5} from "@blueprintjs/core";
+import * as AST from "ast_wrapper";
+import {AppStore, DialogStore, HelpType} from "stores";
+import {DistanceMeasuringStore, RegionCoordinate} from "stores/Frame";
+import {CustomIcon} from "icons/CustomIcons";
+import {SafeNumericInput, ColorPickerComponent, CoordinateComponent} from "components/Shared";
+import {DraggableDialogComponent} from "components/Dialogs";
+import {WCSPoint2D, Point2D} from "models";
+import {getPixelValueFromWCS, getFormattedWCSPoint, isWCSStringFormatValid, SWATCH_COLORS} from "utilities";
+import {ImageViewLayer} from "components";
+import "./DistanceMeasuringDialog.scss";
+
+const KEYCODE_ENTER = "Enter";
+
+@observer
+export class DistanceMeasuringDialog extends React.Component {
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+    }
+
+    @observable WCSMode: boolean = true;
+
+    @action setWCSMode = (bool?: boolean) => {
+        this.WCSMode = bool === undefined ? !this.WCSMode : bool;
+    };
+
+    private handleChangeWCSMode = (formEvent: React.FormEvent<HTMLInputElement>) => {
+        const WCSMode = formEvent.currentTarget.value === RegionCoordinate.Image ? false : true;
+        this.setWCSMode(WCSMode);
+    };
+
+    private handleValueChange = (
+        event: React.KeyboardEvent<HTMLInputElement> | React.FocusEvent<HTMLInputElement>,
+        distanceMeasuringStore: DistanceMeasuringStore,
+        wcsInfo: AST.FrameSet,
+        WCSStart: WCSPoint2D,
+        WCSFinish: WCSPoint2D,
+        isX: boolean,
+        finish?: boolean,
+        pixel?: boolean
+    ) => {
+        const target = event.target as HTMLInputElement;
+
+        if (pixel) {
+            const value = parseFloat(target.value);
+            if (!isFinite(value)) return;
+            if (isX && finish) {
+                distanceMeasuringStore?.setFinish(value, distanceMeasuringStore?.finish.y);
+            } else if (finish) {
+                distanceMeasuringStore?.setFinish(distanceMeasuringStore?.finish.x, value);
+            } else if (isX) {
+                distanceMeasuringStore?.setStart(value, distanceMeasuringStore?.start.y);
+            } else {
+                distanceMeasuringStore?.setStart(distanceMeasuringStore?.start.x, value);
+            }
+        } else if (wcsInfo) {
+            const value = target.value;
```

Comment:
you only use `value` as a string, so rather do the `as string` conversion here

Reply:
resolved